### PR TITLE
docs: adds title and description meta tags via react-helmet

### LINKED ISSUE #4704: docs: Add a <title> tag to each page
## Is your feature request related to a problem? Please describe

Currently, the docs pages just have the URL as the `<title>`, which makes it inconvenient to have multiple tabs open.

## Describe the solution you'd like

I did a quick search and it seems like we should either use `react-helmet` (shown [here in the Gatsby docs](https://www.gatsbyjs.org/docs/add-page-metadata/)) or go a step further and [add an SEO component](https://www.gatsbyjs.org/docs/add-seo-component/).

## Describe alternatives you've considered

Not too familiar with Gatsby, so there might be another way to do this that I'm missing. In particular, we might be able to conveniently do it off of the highest priority header in the Markdown, but I'm not sure.

## Additional context

I don't mind taking this up in a PR. Has there been work done on this before, and is there anything I should be looking out for?

### PR DESCRIPTION
**Pull request by malsf21**
Hey, this is a PR that attempts to fix #4704 by adding an SEO component with `react-helmet`, which creates a `<title>` tag and several SEO-related `<meta>` tags. It supports dynamically changing the title based on whatever page is rendered.

Right now, it changes the title tag to "React-Bootstrap · React-Bootstrap Documentation" - which is not great, but not bad. I'm wondering if there's a good way to grab the actual "title" for each page (short of going through each page and manually inputting one) - for example, is there a good way to grab the value inside the `<h1>` of each page? I'm open to modifying the PR in whatever way is needed to make this all work!

### CODE REVIEW

**Comment by mxschmitt** [comment]
Closed because of stale. If you have a fix for that feel free to submit a new PR.

**Comment by malsf21** [comment]
Hey! I personally am still unsure of how to proceed, but I think it still improves the current functionality (which is nothing in the title, i.e. still mergeable) - do you have any suggestions on what to do?

**Comment by taion** [comment]
oops, i think this is still relevant... @jquense do you have any feedback on whether this is the right way to go?

**Comment by malsf21** [comment]
In the meantime, quickly fixed the merge conflict for `package.json` and `yarn.lock` for the `www/` docs folder. `react-helmet` version might be out of date since I made this PR quite a bit of time ago.

**Comment by malsf21** [comment]
(also as a quick note, I just realised that `upstream/master` actually had merge conflicts in `yarn.lock`, so I ended up resolving them in this PR. if this is a bad side effect, I can undo those commits)

**Comment by malsf21** [comment]
bump @jquense ?

**Comment by malsf21** [comment]
(Due to recent events, I have a bit more dev time on my hands, so I don't mind working significantly on this feature!)

**Comment by malsf21** [comment]
Just going to re-ping again - not sure if any of the core team have any thoughts on this change. As it is, it should still be an improvement over the existing featureset, though there is likely a better way to implement this.

**Comment by kyletsang** [comment]
Hey @malsf21, sorry for delayed response.  I think there's still some interest here, but I'm also unsure of what's the best way.

@jquense thoughts?

**Comment by jquense** [comment]
Seems ok to me! It is an improvement

**Comment by malsf21** [comment]
Okay, going to resolve the merge conflict then, and try to get it ready to merge - still not very familiar with Gatsby, but may try to poke around and figure out a creative solution.

**Comment by malsf21** [comment]
Had to do a bit of wrangling with the branches (didn't realize force-pushing auto-closed the PR), but I think this should work now! Let me know what y'all think!

File: www/src/layouts/index.js
Comment (kyletsang) [inline www/src/layouts/index.js:19]: Can we group this with the other imports?

**Review by kyletsang — APPROVED** [review]
Looks alright to me

File: www/src/layouts/index.js
Comment (malsf21) [reply www/src/layouts/index.js:19]: Yup, sorry - didn't catch that when resolving the merge conflict, will update!

**Comment by malsf21** [comment]
Also - I'm sure y'all are very busy with updating things to v5, but should I submit an issue for more in-depth title + description configuration? Or does this suffice for now?

**Comment by kyletsang** [comment]
I'm ok with breaking things up if you want to do that.  This PR is already an improvement to docs.

**Comment by malsf21** [comment]
Hm the failed checks are because of a brew error (I just resolved a merge conflict)

**Comment by kyletsang** [comment]
> 
> 
> Hm the failed checks are because of a brew error (I just resolved a merge conflict)

There's an issue with the macOS CI right now.  See #5490

This PR good to go?

**Comment by malsf21** [comment]
I think so - still waiting on one more reviewer though!

**Review by jquense — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks!

**Pull request merged**
